commonui/src/main/java/com/hyperwallet/android/ui/common/util/CurrencyParser.java

```diff
@@ -127,4 +128,13 @@ public CurrencyDetails getCurrency(String currencyCode) {
         }
         return null;
     }
+
+    public static String getRateWithFourDecimal(String rate)
+    {
+        NumberFormat nf = NumberFormat.getNumberInstance();
+        nf.setMaximumFractionDigits(4);
```

Comment:
Check for numberformatexception

Reply:
NumberFormat/Decimal formate rounding the value 1.0000 as 1 , using the sub string to achieve that.